packages/typespec-go/src/emitter.ts

```diff
@@ -28,10 +28,56 @@ import { EmitContext, NoTarget } from '@typespec/compiler';
 import 'source-map-support/register.js';
 import { reportDiagnostic } from './lib.js';
 import { CodegenError } from '../../codegen.go/src/errors.js';
+import { execSync } from 'child_process';
 
 export async function $onEmit(context: EmitContext<GoEmitterOptions>) {
   try {
     await generate(context);
+
+    // probe to see if Go tools are on the path
+    try {
```

Comment:
I split this up into separate `try/catch` blocks so that each task has a different `code`. Maybe that's not so useful thought?

Reply:
Will leave as-is for now. Can revisit later.